.cbmc-batch/jobs/string/string_tests.c

```diff
@@ -0,0 +1,198 @@
+#include <aws/common/string.h>
+#include "cbmcproof/make_common_datastructures.h"
+#include "cbmcproof/proof_allocators.h"
+#include <stddef.h>
+
+const size_t MAX_STRING_LEN = 16;
+
+void assert_bytes_match(const uint8_t* a, const uint8_t* b, size_t len)
+{
+  if (len > 0){
+    size_t i;
+    __CPROVER_assume(i < len);
+    assert(a[i] == b[i]);
+  }
+}
+
+void assert_all_bytes_are(const uint8_t* a, const uint8_t c, size_t len)
+{
+  if (len > 0){
+    size_t i;
+    __CPROVER_assume(i < len);
+    assert(a[i] == c);
+  }
+}
+
+void assert_all_zeroes(const uint8_t* a, size_t len)
+{
+  assert_all_bytes_are(a, 0, len);
+}
+
+void aws_string_bytes_harness()
+{
+  struct aws_string* str = make_arbitrary_aws_string_nondet_len(can_fail_allocator());
+  assert(aws_string_bytes(str) == str->bytes);
+}
+
+void aws_string_eq_harness()
+{
+  struct aws_string* str_a = make_arbitrary_aws_string_nondet_len_with_max(can_fail_allocator(), MAX_STRING_LEN);
+  struct aws_string* str_b = nondet_bool()
+    ? str_a : make_arbitrary_aws_string_nondet_len_with_max(can_fail_allocator(), MAX_STRING_LEN);
+  bool rval = aws_string_eq(str_a, str_b);
+  if(rval) {
+    assert(str_a->len == str_b->len);
+    assert_bytes_match(str_a->bytes, str_b->bytes, str_a->len);
+  }
+}
+
+void aws_string_eq_byte_cursor_harness()
+{
+  struct aws_string* str = make_arbitrary_aws_string_nondet_len_with_max(can_fail_allocator(), MAX_STRING_LEN);
+  struct aws_byte_cursor cursor = nondet_bool() ? aws_byte_cursor_from_string(str)
+    : make_arbitrary_byte_cursor_nondet_len_max(MAX_STRING_LEN);
+  bool rval = aws_string_eq_byte_cursor(str, &cursor);
+  if(rval) {
+    assert(str->len == cursor.len);
+    assert_bytes_match(str->bytes, cursor.ptr, str->len);
+  }
+}
+
+void aws_string_eq_byte_buffer_harness()
```

Comment:
We should use the actual name of the function + harness. In this case, replace `aws_string_eq_byte_buffer_harness` with `aws_string_eq_byte_buf`, right?